### PR TITLE
ctb: Add deployDeputyGuardian

### PR DESCRIPTION
**Pull request by maurelian**
Adds the DeputyGuardianModule to the deploy scripts.

### CODE REVIEW

**Comment by maurelian** [comment]
* **#10224** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/10224?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#10207** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/10207?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#10120** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/10120?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#10072** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/10072?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `develop`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @maurelian and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: packages/contracts-bedrock/scripts/Deploy.s.sol [hunk 1]
```diff
@@ -517,18 +548,22 @@ contract Deploy is Deployer {
...
         // Now that the owners have been added and the threshold increased we can deploy the liveness module (otherwise
         // constructor checks will fail).
-        address module = deployLivenessModule();
+        address livenessModule = deployLivenessModule();
 
         // Unfortunately, a threshold of owners is required to actually enable the module, so we're unable to do that
         // here, and will settle for logging a warning below.
+        // _callViaSafe({ _safe: safe, _target: address(safe), _data: abi.encodeCall(ModuleManager.enableModule,
+        // (module)) });
+        // console.log("LivenessModule enabled on SecurityCouncilSafe");
```
Comment (mds1) [inline packages/contracts-bedrock/scripts/Deploy.s.sol]: Delete these commented out lines of code?

Comment (maurelian) [reply packages/contracts-bedrock/scripts/Deploy.s.sol]: TBH I thought they might be helpful for expressing the necessary calls, but I think the WARNING message below is pretty informative.

**Review by mds1 — APPROVED** [review]

File: packages/contracts-bedrock/scripts/DeployOwnership.s.sol [hunk 2]
```diff
@@ -121,13 +126,38 @@ contract DeployOwnership is Deploy {
...
+        save("DeputyGuardianModule", addr_);
+        console.log("New DeputyGuardianModule deployed at %s", addr_);
+    }
+
     /// @notice Deploy a Security Council with LivenessModule and LivenessGuard.
     function deployAndConfigueSecurityCouncilSafe() public returns (address addr_) {
         Safe safe = Safe(payable(deploySafe("SecurityCouncilSafe")));
 
+        vm.startBroadcast();
+        address deputyGuardianModule = deployDeputyGuardianModule();
```
Comment (mds1) [inline packages/contracts-bedrock/scripts/DeployOwnership.s.sol:163]: `deployDeputyGuardianModule` uses `makeAddr("SuperchainConfig")`—which is not a real superchain config—as the superchain config address for the broadcast. I think it should instead get the address of what `deploySuperchainConfig` deployed?

Comment (maurelian) [reply packages/contracts-bedrock/scripts/DeployOwnership.s.sol:163]: That works too, can do

File: packages/contracts-bedrock/scripts/DeployOwnership.s.sol [hunk 3]
```diff
@@ -121,13 +126,38 @@ contract DeployOwnership is Deploy {
...
+        _callViaSafe({
+            _safe: safe,
+            _target: address(safe),
+            _data: abi.encodeCall(ModuleManager.enableModule, (deputyGuardianModule))
```
Comment (mds1) [inline packages/contracts-bedrock/scripts/DeployOwnership.s.sol:167]: Should we add `deputyGuardianModule` to the `SecurityCouncilConfig` struct, then load the example config to get the address we use here? Currently the `SecurityCouncilConfig` is incomplete

**Pull request merged**